src/js/utils/dom.js

```diff
@@ -49,6 +49,38 @@ export function htmlToParentElement(html) {
     return parsedElement;
 }
 
+function supportsHtmlParsing() {
+    // Firefox/Opera/IE throw errors on unsupported types
+    try {
+        // WebKit returns null on unsupported types
+        if (parser.parseFromString('', 'text/html')) {
+            // text/html parsing is natively supported
+            return true;
+        }
+    } catch (err) {/* noop */}
+    return false;
+}
+
+function domParse(html) {
+    if (!parser) {
+        parser = new DOMParser();
+        useDomParser = supportsHtmlParsing();
+    }
+    if (useDomParser) {
+        return parser.parseFromString(html, 'text/html').body;
+    }
+    const doc = document.implementation.createHTMLDocument('');
+    if (html.toLowerCase().indexOf('<!doctype') > -1) {
+        // eslint-disable-next-line no-unsanitized/property
+        doc.documentElement.innerHTML = html;
+    } else {
+        // eslint-disable-next-line no-unsanitized/property
+        doc.body.innerHTML = html;
+    }
```

Comment:
Never seen these objects/methods before! this is some _awesome_ polyfilling. 👏👏

Reply:
Yeah. Adapted from the example at https://developer.mozilla.org/en-US/docs/Web/API/DOMParser